include/tvm/runtime/c_runtime_api.h

```diff
@@ -114,6 +114,7 @@ typedef enum {
   kTVMNNVMLast = 20U,
   // The following section of code is used for non-reserved types.
   kTVMExtReserveEnd = 64U,
+  kTVMBFloat = 65U,
```

Comment:
We do not want BFloat to be passed as PackedFunc argument, most packedfunc argument should always be passed as double

Reply:
I suppose TVM should support kernel generation, e.g. generating a fused "conv+bn+relu", rather than generating end-to-end model, which is the usual case. In this case, we might select some intermediate layers of the model and let TVM generate the selected layers. The layers may require bf16 as the dtype, as they are in the middle of the model.

What I want to say is that we sometimes need bf16 as the input dtype.  In our usecase in Intel, we need to generate a bf16 kernel (e.g. conv+bn+relu).

Reply:
Such dtype is covered by allocating a DLTensor with type_code equals kBFloat, and does not need patch to the code here(needed for parameter argument passing PackedFunc).

Reply:
The particular code is used when we directly pass a constant into PackedFunc, e.g. `f(1.0, some_float_value)`. in these cases double can be used.

Reply:
If we remove this type from TVM runtime, we cannot pass a bf16 array to TVM via Python and users can only pass bf16 buffers via C runtime (or in some awkward way to construct a bf16 DLTensor via Python). Currently, with  kTVMBFloat defined, we can:

```python
        A = te.placeholder((32, ), dtype='bfloat16')
        B = te.placeholder((32, ), dtype='bfloat16')
        d = te.compute((32, ), lambda x: A[x] + B[x])
        sch = te.create_schedule(d.op)
        module = tvm.build(sch, [A, B, d])
        npa = np.random.rand(32).astype('float32')
        npb = np.random.rand(32).astype('float32')
        a_ = np_float2tvm_bf16(npa)
        b_ = np_float2tvm_bf16(npb)
        c_ = tvm.nd.empty((32,), 'bfloat16')
        module(a_, b_, c_)
```

Which is useful for testing and prototyping.

Reply:
I don't think you will `kTVMBFloat` to support this feature. The kDLBFloat flag in the runtime::DataType should be sufficient for NDArray contents(because the type code in the NDArray contents diverges from the TVM type code above the OpaqueHandle).

Reply:
ok I understand. will change that